libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/controller/CameraStateController.kt

```diff
@@ -1,21 +1,46 @@
 package com.mapbox.navigation.ui.app.internal.controller
 
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
 import com.mapbox.navigation.ui.app.internal.Action
 import com.mapbox.navigation.ui.app.internal.State
 import com.mapbox.navigation.ui.app.internal.Store
 import com.mapbox.navigation.ui.app.internal.camera.CameraAction
 import com.mapbox.navigation.ui.app.internal.camera.CameraState
 import com.mapbox.navigation.ui.app.internal.camera.TargetCameraMode
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import kotlinx.coroutines.Dispatchers
 
 @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
 class CameraStateController(
-    store: Store
+    private val store: Store,
 ) : StateController() {
     init {
         store.register(this)
     }
 
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        super.onAttached(mapboxNavigation)
+
+        // using immediate dispatcher causes nested store updates
+        store.select { it.navigation }.observe(Dispatchers.Main) { navigationState ->
```

Comment:
Since https://github.com/mapbox/mapbox-navigation-android/pull/6455 landed, do  we still need the `Dispatchers.Main`. 
@Zayankovsky

Reply:
No, I removed it.